matador/utils/chem_utils.py

```diff
@@ -643,3 +645,36 @@ def get_formula_from_stoich(stoich, elements=None, tex=False, sort=True, latex_s
                 else:
                     form += elem[0] + str(int(elem[1]))
     return form
+
+def magres_reference_shifts(
+    magres: list,
+    reference: dict,
+    shielding_key: str = "chemical_shielding_iso",
+    shift_key: str = "chemical_shift_iso",
+):
+    """ Convert chemical shielding to chemical shift.
+
+    Parameters:
+        magres (list): list of containing magres data.
+        reference (str): dictionary of conversion values in the form {element: [gradient, constant]}.
+
+    Keyword arguments:
+        shielding_key (str): the data key for which the shielding is stored under.
+        shift_key (str): the data key for which the shift will be stored under.
+    """
+
+    relevant_species = list(reference.keys())
+    shielding_key_plural = ''.join([shielding_key, 's'])
+    shift_key_plural = ''.join([shift_key, 's'])
+
+    chemical_shifts = []
+    for ind, species in enumerate(magres["atom_types"]):
+        if species in relevant_species:
+            chemical_shift = (magres[shielding_key_plural][ind] * reference[species][0]) + reference[species][1]
+            chemical_shifts.append(chemical_shift)
+        else:
+            chemical_shifts.append(None)
```

Comment:
Need to check that the shifts haven't already been set for other species:

```suggestion
    chemical_shifts = magres.get(shielding_key_plural, len(magres["atom_types"]) * [None])
    for ind, species in enumerate(magres["atom_types"])
        if species in reference:
            chemical_shift[ind] = (magres[shielding_key_plural][ind] * reference[species][0]) + reference[species][1]
```